routers/api/v1/api.go

```diff
@@ -172,6 +172,9 @@ func reqToken() macaron.Handler {
 		if true == ctx.Data["IsApiToken"] {
 			return
 		}
+		if ctx.Context.IsBasicAuth {
+			ctx.CheckForOTP()
```

Comment:
Does this need a return now? Otherwise it will fall through to the next and may write again.

Reply:
good catch. I've just added it back in.